Increasing height of station 5 of Muon Filter

antonioiuliano2 opened the pull request:
Dear all,
just a small change. Muon filter station 5 should be at the same height of the last three stations, in order to match the TP design. See display here:
![display_muonfilter](https://user-images.githubusercontent.com/18480751/104461665-ec7a2e00-55af-11eb-85d2-99f24f8845a7.png)


@ThomasRuf  and I are currently discussing the addition of the support, to avoid a levitating muon filter.

olantwin reviewed (approved):
I'm surprised that `dy` is updated after placing each upstream station.

Maybe it would be better to update it at the start of each loop?

In that case, line 278 would also become superfluous.

Could you please check the formatting? E.g. spaces around the minus, and I generally would prefer having the  `dy = ...` statement on a separate line so that there is only one statement per line. The whitespace/indent also seems quite strange.

antonioiuliano2 commented:
Dear @olantwin,

thank you for your comments. I have updated dy at the start of the first loop, and removed it from the second, as you suggsted.

I have also improved the formatting within the first loop. Of course, whitespace and formatting should be checked in all the classes, probably with the help of an automatic tool. However, if possible I would prefer to do so in a separate pull request, in order to have a more understable Git history.

olantwin commented:
> thank you for your comments. I have updated dy at the start of the first loop, and removed it from the second, as you suggsted.

Thanks!

> I have also improved the formatting within the first loop. Of course, whitespace and formatting should be checked in all the classes, probably with the help of an automatic tool. However, if possible I would prefer to do so in a separate pull request, in order to have a more understable Git history.

I completely agree. For now I would suggest trying to make sure new code or modified code has reasonable formatting, and maybe at some point we can have a big spring-cleaning.

I will also look into checking commits automatically with `clang-format` and `black` (for C++ and python respectively) when the more urgent tasks are taken care of.

olantwin commented on shipLHC/MuFilter.cxx:
```diff
@@ -241,11 +241,15 @@ void MuFilter::ConstructGeometry()
-		dy = dz * TMath::Tan(TMath::DegToRad() * fSlope);
+	{
+	  dz = (fFeBlockZ + fUpstreamDetZ)*l;
+	  dy = dz * TMath::Tan(TMath::DegToRad() * fSlope);
```
In this case it would probably make things less readable, but one could also use a ternary statement here:
```cpp
dy = (l == fNUpstreamPlanes - 1) ? 
    ShiftYEnd - fShiftY : 
    dz * TMath::Tan(TMath::DegToRad() * fSlope);
```

antonioiuliano2 replied:
Thank you for the suggestion, but I personally prefer the readability of the current structure. It is easier to understand and modify at a later time, if needed.

olantwin replied:
Completely agree in this case. Just wanted to make sure you are aware of it, as it's very useful for conditional initialisation/definition of variables, and rarely taught.
